src/plugins/defoldcollection/defoldcollectionplugin.cpp

```diff
@@ -0,0 +1,444 @@
+                        cell_h["v_flip"] = cell.flippedVertically() ? 1 : 0;
+                        cells.append(replaceTags(QLatin1String(cell_t), cell_h));
+                    }
+                }
+                layer_h["cells"] = cells;
+                layers.append(replaceTags(QLatin1String(layer_t), layer_h));
+                layerOrder++;
+            }
+            map_h["layers"] = layers;
+            map_h["material"] = "/builtins/materials/tile_map.material";
+            map_h["blend_mode"] = "BLEND_MODE_ALPHA";
+            // Below, we input a value that's not necessarily correct in Defold, but it lets the user know what tilesource to link this tilemap with manually.
+            // However, if the user keeps all tilesources in /tilesources/ and the name of the tilesource corresponds with the name of the tileset in Defold,
+            // the value will be automatically correct.
+            map_h["tile_set"] = "/tilesources/" + tileset->name() + ".tilesource";
+
+            QString result = replaceTags(QLatin1String(map_t), map_h);
+            Tiled::SaveFile mapFile(tilemapFilePath);
+            if (!mapFile.open(QIODevice::WriteOnly | QIODevice::Text)) {
+                mError = tr("Could not open file for writing.");
+                return false;
+            }
+            QTextStream stream(mapFile.device());
+            stream << result;
+
+            if (mapFile.error() != QFileDevice::NoError) {
+                mError = mapFile.errorString();
+                return false;
+            }
+
+            if (!mapFile.commit()) {
+                mError = mapFile.errorString();
+                return false;
+            }
+        }
+
+        QVariantHash collection_h;
+        collection_h["children"] = QString("");
+        collection_h["embedded-instances"] = QString("");
+        // optional custom properties that allow exporting a map with an offset
+        collection_h["pos-x"] = map->property(QLatin1String("x-offset")).toInt();
+        collection_h["pos-y"] = map->property(QLatin1String("y-offset")).toInt();
+        collection_h["components"] = components;        
+
+        QString result = replaceTags(QLatin1String(collection_t), collection_h);
+        Tiled::SaveFile mapFile(collectionFile);
+        if (!mapFile.open(QIODevice::WriteOnly | QIODevice::Text)) {
+            mError = tr("Could not open file for writing.");
+            return false;
+        }
+        QTextStream stream(mapFile.device());
+        stream << result;
+
+        if (mapFile.error() != QFileDevice::NoError) {
+            mError = mapFile.errorString();
+            return false;
+        }
+
+        if (!mapFile.commit()) {
+            mError = mapFile.errorString();
+            return false;
+        }
+    } else {
+        // if there are group layers, each group layer is a gameobject, parented to the top gameobject, while being a parent to multiple tilemap components
```

Comment:
Do you think it would be possible to remove this big distinction between "if there are group layers" and "if there are no group layers"? It seems weird to me, that if I group some of my tile layers, that it will go into the group layers branch and ignore any top-level tile layers.

Reply:
Good point, I actually hadn't thought that someone might want to mix the two types of structures. I'll see if I can remove the distinction.